block/block.md

```diff
@@ -1 +1,99 @@
 # Block
+
+## Abstract
+
+The block manager is a key component of full nodes that is responsible for block production or block syncing depending on the node type. Block syncing in this context includes retrieving the published blocks from the network, validating them to raise fraud proofs upon validation failure, and updating the state and storing the validated blocks. A full node invokes multiple of the block manager functionalities in parallel, such as:
+
+* block production (only for sequencer full nodes)
+* block publication to DA network
+* block retrieval from DA network
+* block retrieval from Blockstore (which retrieves blocks from the p2p network)
+* block syncing
+
+## Protocol/Component Description
+
+The block manager is initialized using several parameters as defined below:
+
+* `signing key`: used for signing the blocks after production
+* `config`: block manager configurations (see config options below)
+* `genesis`: initialize the block manager with genesis state
+* `localstore`: store blocks and states
+* `(mempool, proxyapp, eventbus)`: for initializing the executor (state transition function). mempool is also used in the manager to check for availability of transactions for lazy block production
+* `dalc`: the data availability light client used to submit and retrieve blocks to DA network  
+* `blockstore`: to retrieve blocks from block store (p2p blocksync)
+
+Block manager configuration options:
+
+* `BlockTime`: time interval used for block production and block retrieval from Blockstore (default: 1 second)
+* `DABlockTime`: time interval used for both block publication to DA network and block retrieval from DA network (default: 15 seconds)
+* `DAStartHeight`: block retrieval from DA network starts from this height
+* `NamespaceID`: 8 `byte` unique identifier of the rollup
+
+### Block Production
+
+When the full node is operating as a sequencer (aka aggregator), the block manager runs the block production logic. There are two modes of block production, which can be specified in the block manager configurations: `normal` and `lazy`.
+
+In `normal` mode, the block manager runs a timer, which is set to the `BlockTime` configuration parameter, and continuously publishes blocks at `BlockTime` intervals.
+
+In `lazy` mode, the block manager starts building a block when any transaction becomes available in the mempool. After the first notification of the transaction availability, the manager will wait for a 1 second timer to finish before finalizing the block, in order to collect as many transactions from the mempool as possible. The block manager also notifies the full node after every lazy block building.
```

Comment:
Any reason why we wait for 1 second specifically? Maybe we should mention the reason here.

Reply:
the reason is "in order to collect as many transactions from the mempool as possible". specifically 1s to be inline with our default blocktime of 1s.

Reply:
added `The 1 second delay is chosen in accordance with the default block time of 1s.` in addition.